packages/react-components/src/lib/hooks/amountInput.ts

```diff
@@ -0,0 +1,49 @@
+import type { NatValue } from '@agoric/ertp/src/types';
+import { AssetKind } from '@agoric/ertp';
+import { parseAsValue, stringifyValue } from '@agoric/web-components';
+import { useState } from 'react';
+
+type Args = {
+  value: NatValue | null;
+  decimalPlaces: number;
+  onChange: (value: NatValue) => void;
+};
+
+export const useAmountInput = ({ value, decimalPlaces, onChange }: Args) => {
+  const amountString = stringifyValue(value, AssetKind.NAT, decimalPlaces);
+
+  const [fieldString, setFieldString] = useState(
+    value === null ? '' : amountString,
+  );
+
+  const handleInputChange: React.ChangeEventHandler<HTMLInputElement> = ev => {
+    // Inputs with type "number" allow these characters which don't apply to
+    // Amounts, just strip them.
+    const str = ev.target?.value
+      ?.replace('-', '')
+      .replace('e', '')
+      .replace('E', '');
+    setFieldString(str);
+
+    try {
+      const parsed = parseAsValue(str, AssetKind.NAT, decimalPlaces);
+      onChange(parsed);
+    } catch {
+      console.debug('Invalid input', str);
```

Comment:
Maybe add an optional `onError` prop?

Reply:
I don't think this is worth bubbling up necessarily. I'm not sure in practice when `parseAsValue` would throw (because we're giving it a NAT here and a string), it's more just so the app doesn't crash worst case.